corehq/apps/domain/auth.py

```diff
@@ -84,19 +85,29 @@ def basicauth(realm=''):
     # stolen and modified from: https://djangosnippets.org/snippets/243/
     def real_decorator(view):
         def wrapper(request, *args, **kwargs):
+            from corehq.apps.locations.middleware import LocationAccessMiddleware
+
             uname, passwd = get_username_and_password_from_request(request)
             if uname and passwd:
                 user = authenticate(username=uname, password=passwd)
                 if user is not None and user.is_active:
                     request.user = user
-                    return view(request, *args, **kwargs)
+                    request.couch_user = CouchUser.from_django_user(user)
+                    # Check LocationAccessMiddleware again, now that the user is authenticated (FB 262139)
+                    location_middleware = LocationAccessMiddleware()
+                    response = location_middleware.process_view(request, view, args, kwargs)
+                    return view(request, *args, **kwargs) if response is None else response
```

Comment:
Are there any other auth types which similarly need this restriction?

Reply:
i believe we need this for api key auth and digest auth as well